### PR TITLE
Sass 1047

### PR DESCRIPTION
**Pull request by BenCosfordHMRC**
Updating test coverage and added in service method for re-use. Updating logging to know what's happening a bit more too. 

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of master?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current master that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of master?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/repositories/EmploymentUserDataRepository.scala [hunk 1]
```diff
@@ -44,9 +47,16 @@ class EmploymentUserDataRepositoryImpl @Inject()(mongo: MongoComponent, appConfi
...
   replaceIndexes = true
-) with Repository with EmploymentUserDataRepository {
+) with Repository with EmploymentUserDataRepository with Logging {
 
-  def create[T](userData: EmploymentUserData)(implicit user: User[T]): Future[Boolean] = collection.insertOne(userData).toFutureOption().map(_.isDefined)
+  def create[T](userData: EmploymentUserData)(implicit user: User[T]): Future[Boolean] = {
+    collection.insertOne(userData).toFutureOption().map(_.isDefined).recover{
+      case e: Exception =>
+        pagerDutyLog(FAILED_TO_CREATE_EMPLOYMENT_DATA, s"[EmploymentUserDataRepositoryImpl][create] Failed to create employment user data. " +
+          s"Error:${e.getMessage} SessionId: ${user.sessionId}")
```
Comment (joeyymoore) [inline app/repositories/EmploymentUserDataRepository.scala:56]: do you need a fullstop between the Error:${e.getMessage} and SessionId for better readability?

File: app/services/EmploymentSessionService.scala [hunk 2]
```diff
@@ -60,6 +60,15 @@ class EmploymentSessionService @Inject()(employmentUserDataRepository: Employmen
     employmentUserDataRepository.find(taxYear, employmentId)
   }
 
+  def getSessionDataAndReturnResult(taxYear: Int, employmentId: String)(result: EmploymentUserData => Future[Result])
+                                   (implicit user: User[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Result] = {
+
+    employmentUserDataRepository.find(taxYear, employmentId).flatMap {
+      case Some(employmentUserData: EmploymentUserData) => result(employmentUserData)
+      case None => Future(Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear)))
```
Comment (joeyymoore) [inline app/services/EmploymentSessionService.scala:68]: do you want a logger in here to say anything about not finding data and redirecting?

File: it/controllers/employment/CheckEmploymentDetailsControllerISpec.scala [hunk 3]
```diff
@@ -16,7 +16,9 @@
 
 package controllers.employment
 
+import models.User
 import models.employment.{AllEmploymentData, EmploymentData, EmploymentSource, Pay}
+import models.mongo.{EmploymentCYAModel, EmploymentDetails, EmploymentUserData}
```
Comment (joeyymoore) [inline it/controllers/employment/CheckEmploymentDetailsControllerISpec.scala:21]: unused import on EmploymentDetails

File: test/config/MockEmploymentUserDataRepository.scala [hunk 4]
```diff
@@ -43,7 +43,14 @@ trait MockEmploymentUserDataRepository extends MockFactory {
   }
   def mockUpdate(employmentUserData: EmploymentUserData, response: Boolean): CallHandler1[EmploymentUserData, Future[Boolean]] = {
```
Comment (joeyymoore) [inline test/config/MockEmploymentUserDataRepository.scala:44]: employmentUserData is never used, can be removed

File: app/services/EmploymentSessionService.scala (hunk 2 above)
Comment (BenCosfordHMRC) [reply app/services/EmploymentSessionService.scala:68]: the find method already logs it

File: app/repositories/EmploymentUserDataRepository.scala (hunk 1 above)
Comment (BenCosfordHMRC) [reply app/repositories/EmploymentUserDataRepository.scala]: Updated

File: it/controllers/employment/CheckEmploymentDetailsControllerISpec.scala (hunk 3 above)
Comment (BenCosfordHMRC) [reply it/controllers/employment/CheckEmploymentDetailsControllerISpec.scala]: Updated

File: test/config/MockEmploymentUserDataRepository.scala (hunk 4 above)
Comment (BenCosfordHMRC) [reply test/config/MockEmploymentUserDataRepository.scala]: Updated

**Pull request merged**